### PR TITLE
Add CLI for route flow counter

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**

### CODE REVIEW

File: clear/main.py [hunk 1]
```diff
@@ -490,6 +492,50 @@ def flowcnt_trap():
...
+@flowcnt_route.command()
+@click.option('--vrf', help='VRF/VNET name or default VRF')
+@click.option('--namespace', '-n', 'namespace', default=None, type=click.Choice(multi_asic_util.multi_asic_ns_choices()), show_default=True, help='Namespace name or all')
+@click.argument('prefix-pattern', required=True)
+def pattern(prefix_pattern, vrf, namespace):
+    """Clear route flow counters by pattern"""
+    command = "flow_counters_stat -c -t route --prefix_pattern {}".format(prefix_pattern)
+    if vrf:
+        command += ' --vrf {}'.format(vrf)
+    if namespace is not None:
```
Comment (Junchao-Mellanox) [inline clear/main.py:521]: add a comment here why don't use "if namespace"

File: config/flow_counters.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    pattern_table = cfgdb.get_table(FLOW_COUNTER_ROUTE_PATTERN_TABLE)
+    for existing_key in pattern_table:
+        if isinstance(existing_key, tuple):
+            existing_prefix = existing_key[1]
+            existing_key = PATTERN_SEPARATOR.join(existing_key)
+        else:
+            _, existing_prefix = extract_route_pattern(existing_key)
+        if not existing_prefix:
+            click.echo('Detect invalid route pattern in existing configuration {}, removing it...'.format(existing_key))
+            cfgdb.set_entry(FLOW_COUNTER_ROUTE_PATTERN_TABLE, existing_key, None)
```
Comment (Junchao-Mellanox) [inline config/flow_counters.py]: ask user if need remove the existing invalid configuration.

File: config/flow_counters.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    :param db: db object
+    :param vrf: vrf string, empty vrf will be treated as default vrf
+    :param max_allowed_match: max allowed match count, $FLOW_COUNTER_ROUTE_MAX_MATCH_FIELD will be used if not specified
+    :param prefix_pattern: route prefix pattern, automatically add prefix length if not specified
+    :param add: True to add/set the configuration, otherwise remove
+    :param yes: Don't ask question if True
+    :return:
+    """
+    check_route_flow_counter_support()
+    try:
```
Comment (Junchao-Mellanox) [inline config/flow_counters.py]: don't check this while removing

File: flow_counter_util/route.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        exit(0)
+
+    support = capabilities.get(FLOW_COUNTER_CAPABILITY_SUPPORT_FIELD)
+    if support is None:
+        print('Waiting for swss to initialize route flow counter capability, please try again later')
+        exit(0)
+
+    if support != 'true':
+        print('Route flow counter is not supported on this platform')
+        exit(0)
```
Comment (Junchao-Mellanox) [inline flow_counter_util/route.py]: exit(1)?

File: scripts/flow_counters_stat [hunk 5]
```diff
@@ -115,6 +121,8 @@ class FlowCounterStats(object):
             table (list): Table data
         """
         if self.args.json:
+            # The first column of the table might have duplicate value, have to
```
Comment (Junchao-Mellanox) [inline scripts/flow_counters_stat]: remove unused comment

File: config/flow_counters.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        if '/' not in prefix_pattern:
+            prefix_pattern += '/' + str(net.prefixlen)
+
+        key = build_route_pattern(vrf, prefix_pattern)
+        for _, cfgdb in db.cfgdb_clients.items():
+            if _try_find_existing_pattern(cfgdb, net, key, yes):
+                entry_data = cfgdb.get_entry(FLOW_COUNTER_ROUTE_PATTERN_TABLE, key)
+                old_max_allowed_match = entry_data.get(FLOW_COUNTER_ROUTE_MAX_MATCH_FIELD)
+                if old_max_allowed_match is not None and int(old_max_allowed_match) == max_allowed_match:
+                    click.echo('The route pattern is already configured, no configuration changed')
```
Comment (keboliu) [inline config/flow_counters.py]: maybe can rephrase like: 'The route pattern already exists, nothing to be changed.'

**Review by keboliu — APPROVED** [review]

**Pull request closed without merging**